modules/node_modules/@ncigdc/components/Oncogrid/oncoGridParams.js

```diff
@@ -97,10 +97,17 @@ export default function ({
   if (observations.length === 0) return null;
 
   // Clean gene & donor data before using for oncogrid.
-  const donorObs = observations.map(o => o.donorId);
-  const geneObs = observations.map(o => o.geneId);
-  donors = donors.filter(d => donorObs.indexOf(d.id) >= 0); // eslint-disable-line fp/no-mutation
-  genes = genes.filter(g => geneObs.indexOf(g.id) >= 0); // eslint-disable-line fp/no-mutation
+
+  const donorObs = new Set();
+  const geneObs = new Set();
+  for (let i = 0; i < observations.length; i += 1) {
+    const observation = observations[i];
+    donorObs.add(observation.donorId);
+    geneObs.add(observation.geneId);
+  }
+
+  donors = donors.filter(d => donorObs.has(d.id)); // eslint-disable-line fp/no-mutation
+  genes = genes.filter(g => geneObs.has(g.id)); // eslint-disable-line fp/no-mutation
```

Comment:
well donors and genes are pretty small compared to the other lists but it doesn't add complexity to the code does it?

still seems clear that you are looking for intersection of donor/gene ids